R/use_extendr.R

```diff
@@ -195,43 +101,57 @@ extendr_module! {{
   return(invisible(TRUE))
 }
 
-#' Creates example R wrappers for Rust functions.
+#' Write templates from `inst/templates`
+#'
+#' `use_rextendr_template()` is a wrapper around `usethis::use_template()` when
+#' it's available and otherwise implements a simple version of `use_template()`.
+#'
+#' @inheritParams usethis::use_template
+#' @inheritParams use_extendr
 #'
-#' Writes simple R wrappers, without which initial compilation of the
-#' package is impossible. Does not require a compiled library.
-#' Can be used as a fallback.
-#' @param pkg_name The name of the package.
-#' @param outfile Determines where to write wrapper code.
-#' @param path Path from which package root is looked up. Used for pretty message formatting.
-#' @param extra_items Character vector or `NULL` containing additional wrappers.
-#' Should be valid R code.
-#' @param quiet Logical scalar indicating whether the output should be quiet (`TRUE`)
-#'   or verbose (`FALSE`).
 #' @noRd
-make_example_wrappers <- function(pkg_name, outfile, path = ".", extra_items = NULL, quiet = FALSE) {
-  roxcmt <- "#'" # workaround for roxygen parsing bug in raw strings
-
-  wrappers_content <- glue::glue(
-    r"(
-    {roxcmt} @docType package
-    {roxcmt} @usage NULL
-    {roxcmt} @useDynLib {pkg_name}, .registration = TRUE
-    NULL
-    )"
-  )
-
-  if (!is.null(extra_items)) {
-    wrappers_content <- glue::glue_collapse(
-      c(wrappers_content, extra_items),
-      sep = "\n"
+use_rextendr_template <- function(template, save_as = template, data = list(), quiet = getOption("usethis.quiet", FALSE)) {
+  if (is_installed("usethis")) {
+    withr::local_options(usethis.quiet = quiet)
```

Comment:
I noticed that {withr} is not in the `Imports`, but it is used in the source. 
I think we should add the reference. 
In the upcoming PR I am relying on {withr} as well, so it is here to stay.

Reply:
I think the argument was that usethis uses withr and so if usethis is installed then withr is also installed. It's a bit shaky, though, since usethis could in theory stop using withr. I'm fine with moving it to `Imports`.

Reply:
If I understand correctly, it should be OK because we depend on {pkgbuild} 
https://github.com/extendr/rextendr/blob/5ee2062d9134084a99e68266271985e360244d6e/DESCRIPTION#L42-L49
and it in turn depends on {withr}
https://github.com/r-lib/pkgbuild/blob/2aaecf838eb023b788b7a31e267ac644d616dae0/DESCRIPTION#L20.

Reply:
I'm not concerned about usethis switching away from withr (it's central to its internal state principles, and I'm also an author over there now, so I'd have a pretty good heads up), but let's just move it to Imports since we implicitly do so anyway. We can probably make use of it in a few other places, as well.